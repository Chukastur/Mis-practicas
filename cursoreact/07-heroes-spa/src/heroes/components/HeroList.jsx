import { useMemo } from "react";
import { getHeroesByPublisher, HeroCard } from "../"

export const HeroList = ({ publisher }) => {

    const heroes = useMemo ( () => getHeroesByPublisher ( publisher ), [ publisher ] );
   
    return (
        <div className="row row-cols-1 row-cols-md-3 g-3 no-gutters h-100">
            {
                heroes.map( heroe => ( 
                    <div className="col" key={heroe.id}>
                        <HeroCard 
                            key= { heroe.id } {...heroe}  
                        />
                    </div>
                ))
            }        
        </div>
    );
};
 