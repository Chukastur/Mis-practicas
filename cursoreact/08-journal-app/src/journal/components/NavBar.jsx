import { LogoutOutlined, MenuOutlined } from "@mui/icons-material"
import { AppBar, Grid, IconButton, Toolbar, Typography } from "@mui/material"
import { useDispatch } from "react-redux";
import { startLogOut } from "../../store";

export const NavBar = ({ drawerWidth = 300 }) => {

  const dispatch = useDispatch ();

  const onLogout = () => {
    dispatch ( startLogOut () ); 
  };
  

  return (
    <AppBar 
      position="fixed"
      sx={{
        width: { sm: `calc(100% - ${ drawerWidth }px)`},
        ml: { sm: `${drawerWidth}px` }
      }}
      >
      <Toolbar>
        <Grid  
          container 
          direction='row'
          alignItems='center'
          sx={{ width: '100%'}}
          >
          <Grid sx={{ display: 'flex', alignItems: 'center' }}>
            <IconButton
              color="inherit"
              edge='start'
              sx={{ mr:2, display: { sm: 'none' } }}
              >
              <MenuOutlined />
            </IconButton> 
          </Grid>  
          <Grid>
             <Typography variant="h6" noWrap component='div'>Journal App</Typography>
          </Grid> 
        </Grid>

        <Grid justifyContent='end'>
            <IconButton 
              onClick={ onLogout }
              color="error"
            >
              <LogoutOutlined />
            </IconButton>
        </Grid>      
      </Toolbar>

    </AppBar>
  );
};
