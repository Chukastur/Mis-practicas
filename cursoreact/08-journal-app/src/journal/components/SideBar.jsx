import { useSelector } from "react-redux";
import { Box, Divider, Drawer, Grid, List, Toolbar, Typography } from "@mui/material";
import { SideBarItem } from "./SideBarItem";

export const SideBar = ({ drawerWidth = 300 }) => {


  const { displayName } = useSelector ( ( state ) => state.auth );
  const { notes } = useSelector ( ( state ) => state.journal );


  return (

    <Box
      component='nav'
      sx={{ width:drawerWidth, flexShrink:0 }}      
    >
      <Drawer
        variant="permanent"
        open
        sx={{ 
          display: { xs: 'none', sm: 'block' },          
          '& .MuiDrawer-paper': { boxSizing: 'border-box', width: drawerWidth  }
        }}
      >
        <Toolbar>
            <Grid>
              <Typography variant="h6" noWrap component='div'>
                { displayName }
              </Typography>      
            </Grid>
        </Toolbar>
        <Divider />

        <List>
          {
            notes.map( note => (
              <SideBarItem 
                key={ note.id } { ...note }                        
              />
            ))
          }
        </List>

      </Drawer>

    </Box>
    
  );      
}; 
