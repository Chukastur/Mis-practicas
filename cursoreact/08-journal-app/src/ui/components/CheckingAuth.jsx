import { CircularProgress, Grid } from "@mui/material";


export const CheckingAuth = () => {
  return (
    <Grid 
      container
      size={ 6 }
      spacing={ 0 }
      direction='column'
      alignItems='center'
      justifyContent='center'
      sx={{ minHeight: '100vh', backgroundColor: 'primary.main', padding: 4 }}  
    >
      <Grid>
        <CircularProgress color='warning' />
      </Grid>
    </Grid>
  );
};
