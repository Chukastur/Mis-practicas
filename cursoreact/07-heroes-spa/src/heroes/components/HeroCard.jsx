import { Link } from "react-router-dom";
import { HeroList } from './HeroList'


const CharactersByHero = ({ alter_ego, characters  }) =>{
    if ( alter_ego === characters) return (<></>);
    return ( <p>{ characters }</p> )
};

export const HeroCard = ({
    id,
    superhero,
    publisher,
    alter_ego,
    first_appearance,
    characters
}) => {

    const heroImageUrl = `/heroes/${id}.jpg`;

    return (
        <div className="col">
            <div className="card h-100 animate__animated animate__fadeIn">

                <div className="row no-gutters h-100">
                    
                    <div className="col-4">
                        <img 
                            src={ heroImageUrl } 
                            className="card-img h-100" 
                            alt={superhero} 
                            style={{objectFit: 'cover', minHeight: '250px'}}
                        />
                    </div>
                    
                    <div className="col-8">

                        <div className="card-body d-flex flex-column">

                            <div>
                                <h5 className="card-title">{ superhero }</h5>
                                <p className="card-text" style={{minHeight: '2em'}}>{ alter_ego }</p> 
                                <div style={{minHeight: '2em'}}>
                                    <CharactersByHero  alter_ego = {alter_ego} characters = { characters }/>     
                                </div>                                                        
                            </div>

                            <div>
                                <p 
                                    className="card-text flex-grow-1">
                                    <small className="text-muted">{ first_appearance }</small>
                                </p>
                            </div>
                        </div>

                    </div>
                </div>

                <div className="card-footer d-flex flex-column">
                    <Link 
                        to={`/hero/${ id }`}
                        className="btn btn-outline-dark btn-sm"
                    >
                        Mas Informacion
                    </Link>      
                </div>                

            </div>    
        </div>
    
    );
};
