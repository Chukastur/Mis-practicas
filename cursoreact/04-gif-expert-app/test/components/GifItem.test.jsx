import { render, screen } from '@testing-library/react';
import { GifItem } from "../../src/components/GifItem"; 

describe ('Pruebas en <Gift Item />', () =>{

    const title = 'Fernando';
    const url   = 'https:///ejemplo.com';

    test ('debe de hacer match con el snapshot', () =>{
        
        const { container } = render ( <GifItem title= { title } url= { url }/>);
        expect ( container ).toMatchSnapshot ();
    })     
    
    test('Debe mostrar la iamgen con el URL y el ALT indicado', () => {

        render ( <GifItem title= { title } url= { url }/>);
        //screen.debug ();
        //console.log (screen.getByRole('img').src);
        expect ( screen.getByRole('img').getAttribute('src')).toBe ( url );
        expect ( screen.getByRole('img').getAttribute('alt')).toBe ( title );
    });

        test('Debe mostrar el titulo en el componente', () => {

        render ( <GifItem title= { title } url= { url }/>);
        expect ( screen.getByText (title) ).toBeTruthy();
    });
});