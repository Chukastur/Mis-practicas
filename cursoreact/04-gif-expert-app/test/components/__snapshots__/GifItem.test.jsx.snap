// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Pruebas en <Gift Item /> debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Fernando"
      src="https:///ejemplo.com"
    />
    <p>
      Fernando
    </p>
  </div>
</div>
`;
